### DISCUSSION TITLE
building du_example

### DISCUSSION THREAD

**Discussion by mehdimoore** [srsran/srsRAN_Project#498] [General Help]
I have recently started invesitgating srsran, and trying to get my head around the project.
I would be grateful if someone advise me how to build the du_example in apps/examples/du?

**Comment by frankist** [comment]
What do you mean by "build"? It compiles for you, right?

Can you briefly summarize what you intend to do?

**Reply by mehdimoore** [reply]
I mean compiling the du_example. I have cloned the srsRAN_Project and followed the [Installation Guide](https://docs.srsran.com/projects/project/en/latest/user_manuals/source/installation.html#installation-guide) to build the project. I can see an executable being generated in `build/apps/gnb` directory. 

For my project, I am initially intereted to be able to run du at first, therefore I am trying to compile and run `du_example` in ` apps/examples/du` directory. Following the instructions in [Installation Guide](https://docs.srsran.com/projects/project/en/latest/user_manuals/source/installation.html#installation-guide) didn't create `du_example` executable. checking makefiles, I tried to compile all examples but I had no success.
 
 To summarise, I need to be able to compile and run the du_example. and I would be grateful if you could guide me through this

**Reply by pgawlowicz** [reply]
You need to have ZMQ or UHD installed and enabled, check here:
https://github.com/srsran/srsRAN_Project/blob/main/apps/examples/du/CMakeLists.txt

then build the project with:
```
git clone https://github.com/srsran/srsRAN_Project.git
cd srsRAN_Project
mkdir build
cd build
cmake ../ -DENABLE_EXPORT=ON -DENABLE_ZEROMQ=ON
make -j`nproc`
``
